src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -113,6 +127,71 @@ def _try_fetch(self) -> Iterable[ExternalLocation]:
         for row in self._fetch(f"SELECT * FROM {self._schema}.{self._table}"):
             yield ExternalLocation(*row)
 
+    def _get_ext_location_definitions(self, missing_locations: list[ExternalLocation]) -> list:
+        tf_script = []
+        cnt = 1
+        for loc in missing_locations:
+            if loc.location.startswith("s3://"):
+                res_name = loc.location[5:].replace("/", "_")[:-1]
+            elif loc.location.startswith("gcs://"):
+                res_name = loc.location[6:].replace("/", "_")[:-1]
+            elif loc.location.startswith("abfss://"):
+                res_name = (
+                    loc.location[loc.location.index("@") + 1 :]
+                    .replace(".dfs.core.windows.net", "")
+                    .replace("/", "_")[:-1]
+                )
+            else:
+                res_name = f"name_{cnt}"
+            script = f'resource "databricks_external_location" "{res_name}" {{ \n'
+            script += f'    name = "{res_name}"\n'
+            script += f'    url  = "{loc.location[:-1]}"\n'
+            script += "    credential_name = databricks_storage_credential.<storage_credential_reference>.id\n"
+            script += "}\n"
+            tf_script.append(script)
+            cnt += 1
+        return tf_script
+
+    def _match_table_external_locations(self) -> tuple[list, list[ExternalLocation]]:
+        external_locations = list(self._ws.external_locations.list())
+        location_path = [_.url.lower() for _ in external_locations]
+        table_locations = self.snapshot()
+        matching_locations = []
+        missing_locations = []
+        for loc in table_locations:
+            if loc.location[:-1].lower() in location_path:
+                matching_locations.append(
+                    [external_locations[location_path.index(loc.location[:-1])].name, loc.table_count]
+                )
+                continue
+            missing_locations.append(loc)
+        return matching_locations, missing_locations
+
+    def save(self) -> str:
```

Comment:
```suggestion
    def save_as_terraform_definitions_on_workspace(self) -> str:
```
remove folder argument from constructor, it doesn't belong there, pass it as the argument for this method.